substrate/frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -214,11 +212,11 @@ pub fn expand_outer_inherent(
 					return false
 				}
 
+				let call = <#unchecked_extrinsic as ExtrinsicCall>::call(ext);
 				#(
 					#pallet_attrs
 					{
-						let call = <#unchecked_extrinsic as ExtrinsicCall>::call(ext);
-						if let Some(call) = IsSubType::<_>::is_sub_type(call) {
+						if let Some(call) = IsSubType::<_>::is_sub_type(&call) {
 							if <#pallet_names as ProvideInherent>::is_inherent(&call) {
 								return true;
 							}
```

Comment:
`EnsureInherentsAreFirst` below should be able to be replaced with some logic in `Executive`. Otherwise we are decoding twice.